packages/client/src/links/wsLink.ts

```diff
@@ -51,6 +55,12 @@ export interface WebSocketClientOptions extends UrlOptionsWithConnectionParams {
    * Triggered when a WebSocket connection is established
    */
   onOpen?: () => void;
+  /**
+   * Triggered when a WebSocket connection encounters an error
+   * Must return true if reconnection should be attempted, false if not
+   * If a provided Promise rejects, reconnection remains allowed
+   */
+  onError?: (evt?: Event) => Promise<boolean> | boolean;
```

Comment:
```suggestion
  /**
   * Triggered when a WebSocket connection encounters an error
   */
  onError?: (evt?: Event) => void;
```

> The second commit builds upon the first and adds the ability to explicitly prevent further reconnection attempts if the application decides that the underlying issue is non-recoverable. Reconnection can still be triggered explicitly with `wsClient.reconnect()` if the application decides that connection should now be possible again.

I don't understand the purpose of this - the error here won't contain any useful information and reconnection is already handled by the `retryDelayFn`